### ISSUE TITLE
Error Test Failure

### ISSUE THREAD

**Issue by twalmsley** [MODLanguage/ruby-interpreter#6]
This test should produce an error but isn't. Presumably this is because the value of `n` is not a number when the `*superclass` says it should be one.
```
*class(*id=n;*name=name;*superclass=num);n=Elliott
```

**Comment by twalmsley** [comment]
@elliottinvent 
If the superclass is `str` and the value of `n` is 10 like this:
```
*class(*id=n;*name=name;*superclass=str);n=10
```
Should we expect an error or should the value be converted to a String?

**Comment by alexdalitz** [comment]
Nice!

I’d vote for converting to string.

**Comment by elliottinvent** [comment]
First one should return an error yes because "Elliott" can't be converted to a number. 

Agree with Alex that the second one should convert 10 to a string since that is possible.

**Comment by alexdalitz** [comment]
Same with other primitives right? Eg True etc

**Comment by elliottinvent** [comment]
I think the only instance where we should convert is a number to a string, since it offers character efficiency benefits. Don't think we can do it for any others but open to ideas.

**Comment by alexdalitz** [comment]
We already have string representations of other primitives.

So why not?

**Comment by twalmsley** [comment]
Current behaviour is that if the `*superclass` is not specified then the values aren't converted, as in:
```
*class(*id=n;*name=name);n=00;n=01;n=000;n=999;n=Elliott

result ->

[
  {
    "name": false
  },
  {
    "name": true
  },
  {
    "name": null
  },
  {
    "name": 999
  },
  {
    "name": "Elliott"
  }
]
```
But if the `*superclass` is `str` then the output is this:
```
[
  {
    "name": "false"
  },
  {
    "name": "true"
  },
  {
    "name": ""
  },
  {
    "name": "999"
  },
  {
    "name": "Elliott"
  }
]
```
When `*superclass` is `num` only the `n=999` case succeeds. Let me know if you want this to change.

**Comment by elliottinvent** [comment]
This looks right to me.

**Comment by alexdalitz** [comment]
We can represent true and false as numbers...

**Comment by elliottinvent** [comment]
Not strictly:

true: 01
false: 00
null: 000

**Comment by twalmsley** [comment]
Do I have more work to do on this? If not it can be closed and the update will be in gem v0.3.6

**Comment by twalmsley** [comment]
Fixed as defined on the [grammar wiki](https://github.com/MODLanguage/grammar/wiki/Class-Supertype-Processing)
